platform/ios/src/UIColor+MGLAdditions.mm

```diff
@@ -2,6 +2,12 @@
 
 @implementation UIColor (MGLAdditions)
 
+static inline unsigned ToByte(CGFloat f)
+{
+    f = MAX(0, MIN(f, 1)); // Clamp
+    return (unsigned)round(f * 255);
```

Comment:
The style specification declares each component’s type as a JSON number, which can be a float. In practice, I don’t think mbgl supports deep color, so the values would get rounded anyways.